dali/core/cuda_stream_pool.cc

```diff
@@ -22,12 +22,13 @@ namespace dali {
 
 CUDAStreamPool::~CUDAStreamPool() {
   Purge();
-  assert(lease_count_ == 0);
 }
 
 CUDAStreamPool::CUDAStreamPool() {
   int num_devices = 0;
-  CUDA_CALL(cudaGetDeviceCount(&num_devices));
+  auto e = cudaGetDeviceCount(&num_devices);
+  if (e != cudaSuccess && e != cudaErrorNoDevice && e != cudaErrorInsufficientDriver)
```

Comment:
What we get for CPU only? `cudaErrorNoDevice`?

Reply:
That depends on how it's realized. If it's a machine without a driver, we get `cudaErrorInsufficientDriver`, if a driver is there, but there are no devices (e.g. by setting `export CUDA_VISIBLE_DEVICES=""`), it's `cudaErrorNoDevice`.

Reply:
Understood.